src/jdbc.rs

```diff
@@ -370,6 +402,24 @@ mod test {
         Ok(())
     }
 
+    #[test]
```

Comment:
Can we add another test, that tests:

```
jdbc:sqlserver://[::1]:1433;database=prisma-demo;user=SA;password=Pr[sm4_Pr]sm4;trustServerCertificate=true;encrypt=true
```

as the input?

Reply:
Also, would this work with the ADO.net strings. Can we use ipv6 addresses in those?

Reply:
This only addresses the issue for jdbc; I can add support for ado.net too tho

Reply:
for completeness sake

Reply:
> Can we add another test, that tests:

That's not a valid JDBC input string, so we can't test it unfortunately. It needs to be escaped properly to work, which we're already testing in our escaping tests.

Reply:
Ok, that's good, so we test 

```
jdbc:sqlserver://[::1]:1433;database=prisma-demo;user=SA;password={Pr[sm4_Pr]sm4};trustServerCertificate=true;encrypt=true
```

already?